src/app/search/search.component.ts

```diff
@@ -0,0 +1,556 @@
+              }
+            });
+
+            // Sort by legislation.
+            this.milestones = _.sortBy(this.milestones, ['legislation']);
+            this.authors = _.sortBy(this.authors, ['legislation']);
+            this.types = _.sortBy(this.types, ['legislation', 'listOrder']);
+            this.projectPhases = _.sortBy(this.projectPhases, ['legislation']);
+          }
+
+          this.currentSearch = this.storageService.state['search'];
+
+          // reload query params from storage
+          if (this.currentSearch && this.storageService.state['search']) {
+            if (this.storageService.state['search'].filterForUI) {
+              this.filterForUI = this.storageService.state['search'].filterForUI;
+              this.setParamsFromFilters(params);
+            }
+
+            if (this.storageService.state['search'].tableParams) {
+              this.tableParams = this.storageService.state['search'].tableParams;
+            }
+
+            if (this.storageService.state['search'].categorizedQuery) {
+              this.categorizedQuery = this.storageService.state['search'].categorizedQuery
+            }
+          }
+
+          if (_.isEqual(this.tableParams, new TableParamsObject())) {
+            this.tableParams = this.tableTemplateUtils.getParamsFromUrl(params, this.filterForURL);
+            this.terms.keywords = this.tableParams.keywords;
+          }
+
+          // todo this will override, need a different check to set this
+          // set default sort
+          this.tableParams.sortBy = '-datePosted,+displayName';
+
+          this.setFiltersFromParams(params);
+          this.updateCounts();
+
+          if (res.documents && res.documents[0].data && res.documents[0].data.meta.length > 0) {
+            this.tableParams.totalListItems = res.documents[0].data.meta[0].searchResultsTotal;
+            this.documents = res.documents[0].data.searchResults;
+          } else {
+            this.tableParams.totalListItems = 0;
+            this.documents = [];
+          }
+
+          this.loading = false;
+          this.setRowData();
+          this._changeDetectionRef.detectChanges();
+        } else {
+          this.loading = false;
+          this.snackBarRef = this.snackBar.open('Error search documents ...', 'RETRY');
+          this.snackBarRef.onAction().subscribe(() => this.onSubmit());
+          this.router.navigate(['/search']);
+          this.loading = false;
+          this._changeDetectionRef.detectChanges();
+        }
+      });
+
+    if (!this.storageService.state['search']) {
+      this.storageService.state['search'] = {};
+    }
+
+
+    if (!this.filterForAPI) {
+      this.categorizedQuery = { name: 'categorized', value: false }
+    }
+
+    this.searchService.getSearchResults(
+      '',
+      'Document',
+      [
+        this.categorizedQuery
+      ],
+      this.storageService.state['search'].tableParams ? this.storageService.state['search'].tableParams.currentPage : 1,
```

Comment:
Are these always going to hit the defaults? Above on line 224, it sets `this.storageService.state['search']` to an empty object.